src/Controls/src/Core/Handlers/Items/iOS/CarouselViewController.cs

```diff
@@ -61,20 +61,43 @@ public override UICollectionViewCell GetCell(UICollectionView collectionView, NS
 		{
 			UICollectionViewCell cell;
 
-			if (ItemsView?.Loop == true && _carouselViewLoopManager != null)
-			{
-				var cellAndCorrectedIndex = _carouselViewLoopManager.GetCellAndCorrectIndex(collectionView, indexPath, DetermineCellReuseId(indexPath));
-				cell = cellAndCorrectedIndex.cell;
-				var correctedIndexPath = NSIndexPath.FromRowSection(cellAndCorrectedIndex.correctedIndex, 0);
-
-				if (cell is DefaultCell defaultCell)
+			if (ItemsView?.Loop == true)
+			{
+				// In iOS 15 and 16, when the ItemsSource of the CarouselView is updated from another page
+				// via the ViewModel (or similar), GetCell is called immediately—while _carouselViewLoopManager
+				// is still null. As a result, an invalid index is passed to base.GetCell, leading to an ArgumentNullException.
+				// 
+				// However, in iOS 17 and 18, GetCell is only called after navigating back to the page containing
+				// the CarouselView. By that time, CarouselViewLoopManager has been properly initialized during
+				// the window attachment, so the issue does not occur.
+				// 
+				// This fix ensures proper handling across all iOS versions by initializing the loop manager
+				// when needed and providing a fallback implementation, making navigation scenarios work consistently.
+				if (_carouselViewLoopManager is null)
 				{
-					UpdateDefaultCell(defaultCell, correctedIndexPath);
+					InitializeCarouselViewLoopManager();
 				}
 
-				if (cell is TemplatedCell templatedCell)
+				if (_carouselViewLoopManager is not null)
+				{
+					var cellAndCorrectedIndex = _carouselViewLoopManager.GetCellAndCorrectIndex(collectionView, indexPath, DetermineCellReuseId(indexPath));
+					cell = cellAndCorrectedIndex.cell;
+					var correctedIndexPath = NSIndexPath.FromRowSection(cellAndCorrectedIndex.correctedIndex, 0);
+
+					if (cell is DefaultCell defaultCell)
+					{
+						UpdateDefaultCell(defaultCell, correctedIndexPath);
+					}
+
+					if (cell is TemplatedCell templatedCell)
+					{
+						UpdateTemplatedCell(templatedCell, correctedIndexPath);
+					}
+				}
+				else
 				{
-					UpdateTemplatedCell(templatedCell, correctedIndexPath);
+					// Fallback to base implementation if manager couldn't be initialized
+					cell = base.GetCell(collectionView, indexPath);
```

Comment:
The fallback completely bypasses loop behavior, could we include a comment indicating it here?